src/services/workshops/WorkshopController.ts

```diff
@@ -0,0 +1,79 @@
+import Workshop, { IWorkshop } from './models/workshop.model';
+import { GetStudent } from '../student/StudentController';
+import { HTTP400Error, HTTP500Error } from '../../utils/httpErrors';
+
+export async function CreateWorkshops(workshop: IWorkshop): Promise<IWorkshop> {
+    return await Workshop.create(workshop).then(res => {
+        return res;
+    }).catch(err => {
+        console.error('an error occurred', err);
+        throw new HTTP500Error('An error occurred when creating workshops');
+    });
+}
+
+// This is pretty bad to do but because this code will likely never be looked at again, 
+// we're just gonna do a a huge ass query that returns all workshops and student data for that booking. (I know, this is really not good....)
+// Basically same as getAllWorkshops but returns student details as well
+export async function GetAllWorkshopsForReports(): Promise<Array<any>> {
+    let workshops = await Workshop.find({}, (err, workshop) => {
+        if (err) {
+            console.error(err);
+            throw new HTTP500Error('An error occurred when getting the workshops');
+        }
+        return workshop;
+    })
+    .lean(); // This alters the returning result from a Mongoose 'Query' into a plain Javascript Object
+
+    // Iterate over workshops and if they have a "current booking", do a lookup to add the student details for that booking
+    let results = workshops.map(async (workshop: any) => {
+        if (workshop.currentBooking) {
+            const { studentId } = workshop.currentBooking;
```

Comment:
How is this working?? In your interfaces you haven't defined current booking under workshop so id imagine this will always return undefined and then we'll skip the following condition every time.
Do we have a waiting list?? If so and we if we want to return student details for the workshop we're going to need to iterate through that but i dont know what you're trying to achieve with this.